docs/standard_transaction.md

```diff
@@ -0,0 +1,153 @@
+
+You may wonder why we add the public key from our derived private key to the original public key when it's already part of the derivation.  This is because we use the synthetic public key to sign for our delegated spends as well.  When you add two public keys, the private key for the resulting public key is the sum of the original private keys.  If we didn't add the original public key then anyone who knew the hidden puzzle could derive the synthetic private key and could then perform delegated spends!  Adding original public key ensures that there is still a secret component of the synthetic private key, even though half of can be known.
+
+This trick is also neat because it allows us to hide the hidden puzzle in a piece of information that was already necessary for the delegated spend.  It's impossible to guess what the hidden puzzle is, even if it's a standard hidden puzzle!  It's even hard to tell if there's a hidden puzzle at all.  This can also contributes to privacy.  For example, if two parties agree to lock up some coins with a hidden puzzle together, you can share pubkeys and verify that information on the blockchain without revealing anything to the network.  Then, if you both agree that the coins *can* be spent with the hidden puzzle if either party is dishonest, you can trustlessly delegated spend the coins to the correct destinations and it's impossible to tell that they are not just normal everyday spends.
+
+We'll look at the code in a moment, but here's a few terms to know before you look at it:
+
+* **hidden puzzle**: a "hidden puzzle" that can be revealed and used as an alternate way to unlock the underlying funds
+* **synthetic key offset**: a private key cryptographically generated using the hidden puzzle and `original_public_key` as inputs
+* **synthetic public key**: the public key (curried in) that is the sum of `original_public_key` and the public key corresponding to `synthetic_key_offset`
+* **original public key**: a public key, where knowledge of the corresponding private key represents ownership of the coin
+* **delegated puzzle**: a delegated puzzle, as in "graftroot", which should return the desired conditions.
+* **solution**: the solution to the delegated or hidden puzzle
+
+## The Chialisp
+
+Here's the full source and then we'll break it down:
+
+```lisp
+(mod
+
+    (SYNTHETIC_PUBLIC_KEY original_public_key delegated_puzzle solution)
+
+    ; "assert" is a macro that wraps repeated instances of "if"
+    ; usage: (assert A0 A1 ... An R)
+    ; all of A0, A1, ... An must evaluate to non-null, or an exception is raised
+    ; return the last item (if we get that far)
+
+    (defmacro assert items
+        (if (r items)
+            (list if (f items) (c assert (r items)) (q . (x)))
+            (f items)
+        )
+    )
+
+    (include condition_codes.clvm)
+    (include sha256tree1.clvm)
```

Comment:
BTW, we should change `sha256tree1` to `sha256tree` everywhere (`sha256tree` used to be a built-in operator in clvm, and was removed since it's easy to write as a function).

Reply:
I'm good with this but I know @matt-o-how has reservations, so I'm going to leave it how it is for now until we also decide to fix the reference code.